addons/mrp/static/src/views/mrp_documents_kanban/helper/file_upload_hook.js

```diff
@@ -0,0 +1,61 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { useBus, useService } from "@web/core/utils/hooks";
+import { FileUploadProgressContainer } from "@mrp/views/mrp_documents_kanban/helper/file_upload_progress_container";
+
+const { useComponent, useEffect, onWillDestroy } = owl;
+const FileUploadProgressContainerKey = "FileUploadProgressContainer";
+
+export function useFileUpload({ onAdded, onLoaded, onError }) {
+    const component = useComponent();
+    if (onAdded) {
+        useBus(component.env.bus, "FILE_UPLOAD_ADDED", onAdded);
+    }
+    if (onLoaded) {
+        useBus(component.env.bus, "FILE_UPLOAD_LOADED", onLoaded);
+    }
+    if (onError) {
+        useBus(component.env.bus, "FILE_UPLOAD_ERROR", onError);
+    }
+}
+
+/**
+ * The given selector will be used to append the progress component.
+ */
+export function useFileUploadProgress(ref, selector, props, options = {}) {
+    const fileUploadService = useService("file_upload");
+    const mainComponentCategory = registry.category("main_components");
+    if (mainComponentCategory.contains(FileUploadProgressContainerKey)) {
+        mainComponentCategory.remove(FileUploadProgressContainerKey);
+    }
```

Comment:
I'm not sure I understand this part ? And I don't know if this is a good practice.